linetools/isgm/utils.py

```diff
@@ -360,29 +347,61 @@ def table_from_complist(complist):
     Special columns: 'name', 'comment', 'logN', 'sig_logN', 'flag_logN'
     See also complist_from_table()
     """
+    key_order = ['RA', 'DEC', 'name', 'z_comp', 'sig_z', 'Z', 'ion', 'Ej',
+                 'vmin', 'vmax','ion_name', 'flag_N', 'logN', 'sig_logN',
+                 'b','sig_b', 'specfile']
+
     tab = Table()
 
-    # mandatory columns
-    tab['RA'] = [comp.coord.ra.to('deg').value for comp in complist] * u.deg
-    tab['DEC'] = [comp.coord.dec.to('deg').value for comp in complist] * u.deg
-    ion_names = []  # ion_names
+    coords = SkyCoord([icomp.coord for icomp in complist])
+    # Basics
+    tab['RA'] = coords.ra
+    tab['DEC'] = coords.dec
+    tab['name'] = [comp.name for comp in complist]
+    tab['vmin'] = u.Quantity([icomp.vlim[0] for icomp in complist])
+    tab['vmax'] = u.Quantity([icomp.vlim[1] for icomp in complist])
+    tab['Z'] = [icomp.Zion[0] for icomp in complist]
+    tab['ion'] = [icomp.Zion[1] for icomp in complist]
+
+    # . attributes
+    for attrib in ['zcomp', 'Ej', 'flag_N', 'logN', 'sig_logN']:
+        values = [getattr(icomp,attrib) for icomp in complist]
+        if isinstance(values[0], u.Quantity):
+            values = u.Quantity(values)
+        tab[attrib] = values
+    # Rename
+    tab.rename_column('zcomp', 'z_comp')
+
+    # Ion names
+    ion_names = []
     for comp in complist:
         if comp.Zion == (-1,-1):
             ion_names += ["Molecule"]
         else:
             ion_names += [ion_to_name(comp.Zion)]
     tab['ion_name'] = ion_names
-    tab['z_comp'] = [comp.zcomp for comp in complist]
-    tab['vmin'] = [comp.vlim[0].value for comp in complist] * comp.vlim.unit
-    tab['vmax'] = [comp.vlim[1].value for comp in complist] * comp.vlim.unit
+
+    # attrib dict
+    for attrib in ['sig_z', 'b', 'sig_b', 'specfile']:
+        try:
+            values = [icomp.attrib[attrib] for icomp in complist]
+        except KeyError:
+            key_order.pop(key_order.index(attrib))
+            pass
+        else:
+            # Quantity
+            if isinstance(values[0], u.Quantity):
+                values = u.Quantity(values)
+            tab[attrib] = values
 
     # Special columns
-    tab['logN'] = [comp.logN for comp in complist]
-    tab['sig_logN'] = [comp.sig_logN for comp in complist]
-    tab['flag_logN'] = [comp.flag_N for comp in complist]
-    tab['comment'] = [comp.comment for comp in complist]
-    tab['name'] = [comp.name for comp in complist]
-    tab['reliability'] = [comp.reliability for comp in complist]
+    for key in ['comment', 'reliability']:
+        if hasattr(complist[0], key):
```

Comment:
instead of being separated?

Reply:
comment and reliability are not required 
so I prefer to separate them.

Reply:
fine.